api/public/v2/tests/test_api_pull_viewset.py

```diff
@@ -201,3 +202,104 @@ def test_retrieve(self, get_repo_permissions):
             "ci_passed": None,
             "author": None,
         }
+
+    @patch("api.shared.permissions.RepositoryArtifactPermissions.has_permission")
+    @patch("api.shared.permissions.SuperTokenPermissions.has_permission")
+    def test_no_pull_if_unauthenticated_token_request(
+        self,
+        super_token_permissions_has_permission,
+        repository_artifact_permisssions_has_permission,
+    ):
+        repository_artifact_permisssions_has_permission.return_value = False
+        super_token_permissions_has_permission.return_value = False
+
+        res = self.client.get(
+            reverse(
+                "api-v2-pulls-detail",
+                kwargs={
+                    "service": self.org.service,
+                    "owner_username": self.org.username,
+                    "repo_name": self.repo.name,
+                    "pullid": self.pulls[0].pullid,
+                },
+            )
+        )
+        assert res.status_code == 403
+        assert (
+            res.data["detail"] == "You do not have permission to perform this action."
+        )
+
+    @override_settings(SUPER_API_TOKEN="testaxs3o76rdcdpfzexuccx3uatui2nw73r")
+    @patch("api.shared.permissions.RepositoryArtifactPermissions.has_permission")
+    def test_no_pull_if_not_super_token_nor_user_token(
+        self, repository_artifact_permisssions_has_permission
+    ):
+        repository_artifact_permisssions_has_permission.return_value = False
+
+        res = self.client.get(
+            reverse(
+                "api-v2-pulls-detail",
+                kwargs={
+                    "service": self.org.service,
+                    "owner_username": self.org.username,
+                    "repo_name": self.repo.name,
+                    "pullid": self.pulls[0].pullid,
+                },
+            ),
+            HTTP_AUTHORIZATION="Bearer 73c8d301-2e0b-42c0-9ace-95eef6b68e86",
+        )
+        assert res.status_code == 401
+        assert res.data["detail"] == "Invalid token."
+
+    @override_settings(SUPER_API_TOKEN="testaxs3o76rdcdpfzexuccx3uatui2nw73r")
+    @patch("api.shared.permissions.RepositoryArtifactPermissions.has_permission")
+    def test_no_pull_if_super_token_but_no_GET_request(
+        self, repository_artifact_permisssions_has_permission
+    ):
+        repository_artifact_permisssions_has_permission.return_value = False
+        res = self.client.post(
+            reverse(
+                "api-v2-pulls-detail",
+                kwargs={
+                    "service": self.org.service,
+                    "owner_username": self.org.username,
+                    "repo_name": self.repo.name,
+                    "pullid": self.pulls[0].pullid,
+                },
+            ),
+            HTTP_AUTHORIZATION="Bearer testaxs3o76rdcdpfzexuccx3uatui2nw73r",
```

Comment:
Optional: since this test succeeds because the token must match the SUPER_API_TOKEN, perhaps make a constant MOCK_SUPER_API_TOKEN up top and use it where needed